crates/networking/discv5/src/discovery.rs

```diff
@@ -54,22 +56,36 @@ struct QueryResult {
 
 pub struct Discovery {
     discv5: Discv5,
+    local_enr: Enr,
     event_stream: EventStream,
     discovery_queries: FuturesUnordered<Pin<Box<dyn Future<Output = QueryResult> + Send>>>,
     find_peer_active: bool,
     pub started: bool,
 }
 
 impl Discovery {
-    pub async fn new(local_key: Keypair, config: &NetworkConfig) -> anyhow::Result<Self> {
-        let enr_local = convert_to_enr(local_key)?;
-        let enr = Enr::builder()
+    pub async fn new(
+        local_key: libp2p::identity::Keypair,
+        config: &NetworkConfig,
+    ) -> anyhow::Result<Self> {
+        let enr_local =
+            convert_to_enr(local_key).map_err(|err| anyhow!("Failed to convert key: {err:?}"))?;
+        let mut enr_builder = Enr::builder();
+        if let Some(socket_address) = config.socket_address {
+            enr_builder.ip(socket_address);
+        }
+        if let Some(socker_port) = config.socket_port {
+            enr_builder.udp4(socker_port);
+        }
+        let enr = enr_builder
             .add_value(ENR_ETH2_KEY, &ENRForkID::pectra())
+            .add_value(ATTESTATION_BITFIELD_ENR_KEY, &config.subnets)
             .build(&enr_local)
-            .map_err(|e| anyhow::anyhow!("Failed to build ENR: {}", e))?;
+            .map_err(|e| anyhow!("Failed to build ENR: {}", e))?;
```

Comment:
```suggestion
            .map_err(|err| anyhow!("Failed to build ENR: {err}"))?;
```

we should inline this